fdbserver/BlobManifest.actor.cpp

```diff
@@ -411,9 +452,16 @@ class BlobManifestDumper : public ReferenceCounted<BlobManifestDumper> {
 				Value versionEncoded = BinaryWriter::toValue(readVersion, Unversioned());
 				splitter->append(KeyValueRef(blobManifestVersionKey, versionEncoded));
 
+				// Inject a marker and check later to make sure the file is encrypted
+				if (g_network->isSimulated() && g_simulator->dataAtRestPlaintextMarker.present()) {
```

Comment:
inject a marker for encryption test.

Reply:
It seems like we are `injecting` marker string and checking if the on-disk file contains the marker string. 

For ApiCorrectness, we updated the generated Key/Value to include the marker at random locations (https://github.com/apple/foundationdb/blob/main/fdbserver/workloads/ApiWorkload.actor.cpp#L242), given randomness it helps in ensuring prefix compression in storage engine like Redwood would not compress the keys and hide the problem if any. Is it possible to follow the same semantics here?

Reply:
My point was: maybe we just update the `key/value generator` and insert marker string there, in someways that would mimic the realistic workload pattern?

Reply:
discussed offline with Ata and Josh about this topic. The marker need to be added to keys in BlobGranulesApiCorrectness. No need to inject marker to manifest files directly.

Reply:
For manifest adding `marker` to the Key is sufficient enough, however, if we are able to add marker to both `key` and `value` it would extend the coverage of BlobGranule encryption overall. 

My understanding is we don't want to update `value` to maintain existing `value corruption` checks in place, an alternate way to overcome this would be suffix `value_buffer_checksum` as part of value itself and then on reads validate if the checksum matches (encryption single token authentication follow this semantics) if possible. If not, then for now we have `BlobGranuleApiCorrectness.toml` to run both ApiCorrectness and BlobGranuleVerifier workloads as part of a simulation run to cover that usecase. Also, if we end up updating value, then might we can remove this workload.

Thanks

Reply:
Thank you Hui for this PR and incorporating change. Sounds like a good plan.